ckanext/cnra_schema/waf_helpers.py

```diff
@@ -0,0 +1,432 @@
+    package_dict['metadataContactAddress'] = metadata_contact.get('contact_address')
+
+    return package_dict
+
+
+def set_waf_data_quality_information(package_dict, iso_values):
+    # (2) Data_Quality_Information
+
+    if iso_values.get('attribute-accuracy-report'):
+        package_dict['attributeAccuracyReport'] = iso_values['attribute-accuracy-report']
+
+    if iso_values.get('completeness-report'):
+        package_dict['completenessReport'] = iso_values['completeness-report']
+
+    if iso_values.get('lineage-citation'):
+        citation = iso_values.get('lineage-citation')
+        lineage_citation = {
+            'title': citation['title'],
+            'originator': citation['origin'],
+            'publicationDate': citation['pubdate'],
+            # 'edition': citation['edition'],
+            'geospatialDataPresentationForm': citation['geoform'],
+            # 'publisher': citation['publish'],
+            # 'onlineLinkage': citation['onlink']
+        }
+        lineage_citation = json.dumps(lineage_citation)
+        package_dict['lineageCitation'] = lineage_citation
+
+    if iso_values.get('source-scale-denominator'):
+        package_dict['sourceScaleDenominator'] = iso_values['source-scale-denominator']
+
+    if iso_values.get('type-of-source-media'):
+        package_dict['typeOfSourceMedia'] = iso_values['type-of-source-media']
+
+    if iso_values.get('source-currentness-reference'):
+        package_dict['sourceCurrentnessReference'] = iso_values['source-currentness-reference']
+
+    if iso_values.get('source-citation-abbreviation'):
+        package_dict['sourceCitationAbbreviation'] = iso_values['source-citation-abbreviation']
+
+    if iso_values.get('distribution-liability'):
+        package_dict['distributionLiability'] = iso_values['distribution-liability']
+
+    return package_dict
+
+
+def set_waf_spatial_reference_information(package_dict, iso_values):
+    ## (4) Spatial_Reference_Information
+
+    if iso_values.get('map-projection-name'):
+        package_dict['mapProjectionName'] = iso_values['map-projection-name']
+
+    if iso_values.get('planar-coordinate-information'):
+        coordinate_information = iso_values.get('planar-coordinate-information')
+        planar_coordinate_information = {
+            'planarCoordinateEncodingMethod': coordinate_information['planar-coordinate-encoding-method'],
+            'abscissaResolution': coordinate_information['abscissa-resolution'],
+            'ordinateResolution': coordinate_information['ordinate-resolution'],
+            'planarDistanceUnits': coordinate_information['planar-distance-units']
+        }
+
+        try:
+            planar_coordinate_information = json.dumps(planar_coordinate_information)
+            package_dict['planarCoordinateInformation'] = planar_coordinate_information
+        except TypeError:
+            print("Unable to serialize the object {}".format(planar_coordinate_information))
```

Comment:
Use log instead of print. The error message can also be improved by indicating what field is causing an issue.
```
"Unable to serialize planarCoordinateInformation: {}".format(planar_coordinate_information)
```

Reply:
As mentioned on this morning sync meeting, the log message will be removed since this is secondary metadata.

Reply:
The print message is still here.